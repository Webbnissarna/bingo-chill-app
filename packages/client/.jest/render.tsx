import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
