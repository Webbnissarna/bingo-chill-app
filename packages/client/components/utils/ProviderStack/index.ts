export { default } from "./ProviderStack";
export * from "./ProviderStack";
