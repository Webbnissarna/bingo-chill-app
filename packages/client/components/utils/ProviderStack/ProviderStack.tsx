"use client";

import DefaultServiceRegistryProvider from "@/services/ServiceRegistry/ServiceRegistryContext";
import { nordThemeColors } from "@webbnissarna/bingo-chill-common/src/utils/theme";
import { ConfigProvider } from "antd";
import type { ReactNode } from "react";

export interface ProviderStackProps {
  children: ReactNode;
}

export default function ProviderStack({
  children,
}: ProviderStackProps): JSX.Element {
  return (
    <DefaultServiceRegistryProvider>
      <ConfigProvider
        theme={{
          token: {
            colorPrimary: nordThemeColors.frost[3],
          },
          components: {
            Upload: {
              actionsColor: "#f00",
              colorBgMask: "#ff0",
              colorBorder: nordThemeColors.polarNight[3],
              colorFillAlter: nordThemeColors.polarNight[0],
              colorPrimary: nordThemeColors.frost[3],
              colorPrimaryHover: "#a80",
              colorText: nordThemeColors.snowStorm[2],
              colorTextDescription: "#e09",
              controlHeightLG: 25,
            },
            Select: {
              clearBg: nordThemeColors.polarNight[3],
              multipleItemBg: nordThemeColors.polarNight[1],
              optionActiveBg: nordThemeColors.polarNight[3],
              optionSelectedBg: nordThemeColors.polarNight[1],
              selectorBg: nordThemeColors.polarNight[0],
              optionSelectedColor: nordThemeColors.snowStorm[0],
              multipleItemBorderColor: nordThemeColors.polarNight[0],
              colorText: nordThemeColors.snowStorm[0],
              colorBgElevated: nordThemeColors.polarNight[0],
              colorTextPlaceholder: nordThemeColors.snowStorm[2],
              colorBorder: nordThemeColors.polarNight[1],
            },
            Table: {
              borderColor: nordThemeColors.polarNight[0],
              headerBg: nordThemeColors.polarNight[0],
              headerColor: nordThemeColors.frost[2],
              headerSplitColor: nordThemeColors.polarNight[3],
              rowHoverBg: nordThemeColors.polarNight[2],
              colorBgContainer: nordThemeColors.polarNight[1],
            },
          },
        }}
      >
        {children}
      </ConfigProvider>
    </DefaultServiceRegistryProvider>
  );
}
