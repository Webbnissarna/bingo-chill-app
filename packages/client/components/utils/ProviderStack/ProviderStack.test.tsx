import { render } from "../../../.jest/render";
import ProviderStack from ".";

describe("ProviderStack", () => {
  it("renders", () => {
    const { asFragment } = render(<ProviderStack>{}</ProviderStack>);

    expect(asFragment()).toMatchSnapshot();
  });
});
