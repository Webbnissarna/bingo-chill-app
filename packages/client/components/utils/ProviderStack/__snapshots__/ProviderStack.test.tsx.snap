// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderStack renders 1`] = `
<DocumentFragment>
  <div
    class="w-screen h-screen max-w-full max-h-full bg-polarNight-0 flex items-center justify-center"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning css-dev-only-do-not-override-18iikkb"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
